### ISSUE TITLE
JOSS review: questions regarding optimization

### ISSUE THREAD

**Issue by fabian-sp** [yhtang/FunFact#243]
This is related to the review of FunFact for JOSS (see https://github.com/openjournals/joss-reviews/issues/4502)

I have a few questions regarding your optimization procedures. 

- If I understand correctly, i could use any `torch.optim` optimizer as `opt` argument, because they possess a `step` method, correct? As I am not so familiar with JAX, is this also true for JAX? Maybe you could provide a short example for this in the docs (or at least I couldn't find it)?
- Regarding the point above, I was wondering why you reimplemented `Adam` and `RMSProp` in https://github.com/yhtang/FunFact/blob/4e5694f7c9881223fcb41fcb21e49007586aa779/funfact/optim.py. They are included in Pytorch, so reimplementing them seems a bit counterintuitive to me. Even though these algorithms are pretty simple, this is maybe an unnecessary source of errors. Why couldn't you use by default Adam from `torch` or some JAX optimizers depending on the active backend?
- A minor point: I noticed that you always call the function `gradient_descent` in https://github.com/yhtang/FunFact/blob/master/funfact/algorithm.py, even though the executed algorithm might not be gradient descent (but e.g. `Adam`). This seems to be slightly misleading naming to me.
